FI-3837: Add CRD Client verifies_requirements

elsaperelli opened the pull request:
# Summary
This PR adds `verifies_requirements` annotations to associated tests in the CRD Client test suite. Any requirements that are not currently tested were added to a Jira ticket for attestation test creation.

# Testing Guidance
- `bundle exec rspec`
- `bundle exec rubocop`
- Run the CRD Client test suite
- Go through the `verifies_requirements` and make sure the test matches the requirement and that it is at the most refined level
- Make sure none of the requirements listed in the Jira ticket (or the attached document) for attestation testing do not have associated tests already implemented

[CRD Client Requirements.docx](https://github.com/user-attachments/files/19373635/CRD.Client.Requirements.docx)

karlnaden reviewed (changes requested):
see [notes](https://github.com/inferno-framework/davinci-crd-test-kit/pull/21#issuecomment-2786446678) for requested changes

elsaperelli commented:
Also unsure of what to do about the `Gemfile.lock` diffs for this one

karlnaden commented:
> Also unsure of what to do about the `Gemfile.lock` diffs for this one

seems pretty safe to resolve with the updated versions. The whole platform stuff is a bit opaque, but I think we can trust that the bundler knows what it is doing.

elsaperelli commented:
- 71: Ah yeah I only found it for the prefetch resources. I will remove it and add it to attestation tests to be made?
- 74: Removed!
- 92: Added!
- 142-145: Didn't think these were tested anywhere- assumed they fell under the payer limitation?
-  284-295: Honestly I don't totally remember, but I think I was looking at the type of the hook and associating it with only the order hooks? Should they go in the `client_fhir_api_group.rb` instead?
- 297, 303: Removed
- 43: Added to FHIR RESTful capabilities (`client_fhir_api_group.rb`) and prefetch test `hook_request_valid_prefetch_test.rb`

karlnaden commented:
> * 71: Ah yeah I only found it for the prefetch resources. I will remove it and add it to attestation tests to be made?

I guess so... this is one of them that it hurts a little bit to be creating an attestation since it is pretty clear how to automatically check it...  But I looked and its not clear exactly where to add it, so attestation (so tempting to try and add it into the [context test](https://github.com/inferno-framework/davinci-crd-test-kit/blob/main/lib/davinci_crd_test_kit/client_tests/hook_request_valid_context_test.rb#L25).

> * 142-145: Didn't think these were tested anywhere- assumed they fell under the payer limitation?

I agree - I put them into an attestation bucket "payer to service map multiple coverage handling", so you can deal with them when you get to creating attestations.

> * 284-295: Honestly I don't totally remember, but I think I was looking at the type of the hook and associating it with only the order hooks? Should they go in the `client_fhir_api_group.rb` instead?

I think they are all being tested in the [crd_hook_request_valid_context test](https://github.com/inferno-framework/davinci-crd-test-kit/blob/main/lib/davinci_crd_test_kit/client_tests/hook_request_valid_context_test.rb#L25) based on that description.  I think you can add specific requirements to instances of those tests like [here](https://github.com/inferno-framework/davinci-crd-test-kit/blob/main/lib/davinci_crd_test_kit/client_tests/client_appointment_book_group.rb#L66):

```
test from: :crd_hook_request_valid_context do
  verifies_requirements '<req>', ....
end
```

elsaperelli commented:
Based on your comment about requirement 71, I tried throwing together a check for context resource IDs...let me know if what I have makes sense, I think it works. Happy to adjust or just pull it out.

karlnaden commented:
> Based on your comment about requirement 71, I tried throwing together a check for context resource IDs...let me know if what I have makes sense, I think it works. Happy to adjust or just pull it out.

Looks good - now just need to associate with a test so that it shows coverage.  Seems like it is checked on the order-sign, order-select, and appointment-book context checks, so I'd add to those 3.

karlnaden reviewed (approved):
add requirement 71 in the few places needed and this is good - no need for a final review
